plugins/inline-color/prism-inline-color.js

```diff
@@ -0,0 +1,90 @@
+(function () {
+
+	if (typeof self === 'undefined' || typeof Prism === 'undefined' || typeof document === 'undefined') {
+		return;
+	}
+
+	// Copied from the markup language definition
+	var HTML_TAG = /<\/?(?!\d)[^\s>\/=$<%]+(?:\s(?:\s*[^\s>\/=]+(?:\s*=\s*(?:"[^"]*"|'[^']*'|[^\s'">=]+(?=[\s>]))|(?=[\s/>])))+)?\s*\/?>/g;
+
+	var HEX_COLOR = /^#?(?=(?:[\da-f]{1,2}){3,4}$)([\da-f][\da-f]?)([\da-f][\da-f]?)([\da-f][\da-f]?)([\da-f][\da-f]?)?$/i;
+
+	/**
+	 * Parses the given hexadecimal representation and returns the parsed RGBA color.
+	 *
+	 * If the format of the given string is invalid, `undefined` will be returned.
+	 * Valid formats are: `RGB`, `RGBA`, `RRGGBB`, and `RRGGBBAA`.
+	 *
+	 * @param {string} hex
+	 * @returns {string | undefined}
+	 */
+	function parseHexColor(hex) {
+		// the regex explained: In first lookahead we check whether the string is valid and then we use the
+		// capturing groups to split the string into its components.
+		var match = HEX_COLOR.exec(hex);
+		if (!match) {
+			return undefined;
+		}
+
+		// This is used to scale normalize 4bit and 8bit values
+		var scale = hex.length <= 4 ? 1 / 15 : 1 / 255;
+
+		var rgb = match.slice(1, 4).map(function (c) {
+			return String(Math.round(parseInt(c, 16) * scale * 255));
+		}).join(',');
+
+		var alpha = match[4] === undefined ? '1' : (parseInt(match[4], 16) * scale).toFixed(3);
+
+		return 'rgba(' + rgb + ',' + alpha + ')';
+	}
+
+	/**
+	 * Validates the given Color using the current browser's internal implementation.
+	 *
+	 * @param {string} color
+	 * @returns {string | undefined}
+	 */
+	function validateColor(color) {
+		var s = new Option().style;
+		s.color = color;
+		return s.color ? color : undefined;
+	}
+
+	/**
+	 * An array of function which parse a given string representation of a color.
+	 *
+	 * These parser serve as validators and as a layer of compatibility to support color formats which the browser
+	 * might not support natively.
+	 *
+	 * @type {((value: string) => (string|undefined))[]}
+	 */
+	var parsers = [
+		parseHexColor,
+		validateColor
+	];
+
+
+	Prism.hooks.add('wrap', function (env) {
+		if (env.type === 'color' || env.type === 'hexcode') {
+			var content = env.content;
+
+			// remove all HTML tags inside
+			var rawText = content.split(HTML_TAG).join('');
+
+			var color;
+			for (var i = 0, l = parsers.length; i < l && !color; i++) {
+				color = parsers[i](rawText);
+			}
+
+			if (!color) {
+				return;
+			}
+
+			var previewElement = '<span class="inline-color" style="background-color:' + color + ';"></span>';
+
+			env.attributes['data-color'] = '';
```

Comment:
Why are we clearing this?

Reply:
We don't clear it, we add it. 
I just need something to identify the elements with inline color preview and didn't want it to be a class because some of our plugins change these.

Reply:
A class like `color-preview` would be an issue? I know the prefixer plugin might change the class names but what else impacts this?

Reply:
It can do a little more than just adding a prefix, but yes, that's my concern.

Reply:
Other plugins create their own elements without running through that plugin (e.g. the Toolbar plugin). I would use a class here & file an issue for running plugin-created elements through that plugin, cuz that's more widespread than just this.

Reply:
So you like an ignore filter for special plugin classes?
This will work but then either all plugins have to communicate their classes in one form or another. What control do you suggest?

---

Also, Toolbar isn't affected because its classes are outside the reach of the `wrap` hook which is the hook [Custom class uses](https://github.com/PrismJS/prism/blob/010a0157b8976ec544b4931f09dd06c1daa89c45/plugins/custom-class/prism-custom-class.js#L58).

Reply:
Oh, I got my understanding backwards–I thought this wasn't run through custom classes & should be, but the issue is it _is_ through custom classes and... shouldn't be? I'm not sure why it shouldn't be though. When using custom classes, you have to modify the theme yourself anyway, as we have no mechanism for producing themes with the customization added. All that would mean is if you use custom classes with this plugin, you have the modify the CSS for this plugin manually as well.

Reply:
Ahhh, let's not make this bigger than it is. Let's use a class. It's only used in the CSS file anyway and I guess you have to expect that using CC will screw up the CSS.

Reply:
Sorry, I didn't see your comment soon enough and I agree.

Reply:
I removed the attribute. Why did I even add it to begin with?